applications/trilinos_application/trilinos_space.h

```diff
@@ -492,7 +492,7 @@ class TrilinosSpace
 
     //***********************************************************************
 
-    static void GatherValues(const VectorType& x, const std::vector<std::size_t>& IndexArray, double* pValues)
+    static void GatherValues(const VectorType& x, const std::vector<int>& IndexArray, DataType* pValues)
     {
```

Comment:
As @pooyan-dadvand said, this function is most likely unused up to now. This is relevant because it contains a bug (it is essentially returning a pointer to a local variable, which is deallocated as soon as the function exits). I think we should remove it, but this is probably a matter for a different PR.